cmd/util/flags/cliflags/job.go

```diff
@@ -0,0 +1,232 @@
+	return emojiString
+}
+
+func SafeAnnotationRegex() *regexp.Regexp {
+	r := regexp.MustCompile(fmt.Sprintf("[^%s|^%s]", returnAllEmojiString(), RegexString))
+	return r
+}
+
+func (j *JobSettings) Labels() (map[string]string, error) {
+	rawLabels := j.labels
+	if j.cmd.Flags().Changed("labels") {
+		rawLabels = make(map[string]string)
+		for _, l := range j.legacy.labels {
+			// NB: the legacy flag is a string slice, but the task expects a map of key,value labels.
+			// We set only the key here.
+			rawLabels[l] = ""
+		}
+	}
+	for k, v := range rawLabels {
+		if !IsSafeAnnotation(k) || k == "" {
+			return nil, fmt.Errorf("label with key %q is not allowed", k)
+		}
+		if !IsSafeAnnotation(v) || v == "" {
+			return nil, fmt.Errorf("label with value %q is not allowed", k)
+		}
+	}
+
+	return rawLabels, nil
+}
+
+type LegacyJobFlags struct {
+	// Deprecated: use `JobSettings.jobType`
+	targetingMode model.TargetingMode
+	// Deprecated: use 'JobSettings.labels'
+	labels []string
+	// Deprecated: use 'JobSettings.constraints'
+	selectors string
+	//Deprecated: use `JobSettings.count`
+	concurrency int
+}
+
+func DefaultJobSettings() *JobSettings {
+	return &JobSettings{
+		name:        "main",
```

Comment:
"main" should be the name of the task and not the job

Reply:
What should be the default name of the job? Seems we still need to set one..

Now that I have merged this with main and compared with the changes made in https://github.com/bacalhau-project/bacalhau/pull/4005 I see we are setting the job name to the ID: https://github.com/bacalhau-project/bacalhau/blob/main/pkg/models/job.go#L166 this Normalization will work on the server side after the job has been submitted and an ID has been created, but the change is causing failures in this branch as the ID isn't set on the client side and Validate thus returns an error: https://github.com/bacalhau-project/bacalhau/blob/main/pkg/models/job.go#L214

How would you like to address this issue?

Reply:
addressed